addOns/client/src/test/java/org/zaproxy/addon/client/internal/ClientSideComponenetUnitTest.java

```diff
@@ -0,0 +1,146 @@
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.zaproxy.addon.client.ExtensionClientIntegration;
+import org.zaproxy.zap.testutils.TestUtils;
+
+/* Unit Tests for {@code ClientSideComponenetUnitTest}
+ * SoretedSets are used for compareTo testing
+ */
+class ClientSideComponenetUnitTest extends TestUtils {
+
+    private static final String EXAMPLE_URL = "https://example.com";
+
+    @BeforeAll
+    static void init() {
+        mockMessages(new ExtensionClientIntegration());
+    }
+
+    /*         zoo.getUserObject()
+    .addComponent(
+            new ClientSideComponent(
+                    Map.of(),
+                    "", // tagName
+                    "fooz", // id
+                    "", // parentUrl
+                    zooUrl, // href
+                    "zShould be ignored", // text
+                    "Cookies", // type
+                    "", // tagType
+                    -1 // formId
+                    )); */
+
+    // typeForDisplay, href, text, id, tagName, tagType, formId
+    @Test
+    void shouldOrderByTypeForDisplayThenHrefThenTextSameHrefs() {
+        // Given
+        ClientSideComponent one =
+                new ClientSideComponent(
+                        Map.of(), "", "foo", "", EXAMPLE_URL, "zNotDisplayed", "Cookies", "", -1);
+        ClientSideComponent two =
+                new ClientSideComponent(Map.of(), "A", "foo", "", EXAMPLE_URL, "zLink", "", "", -1);
+        ClientSideComponent three =
+                new ClientSideComponent(
+                        Map.of(), "", "foo", "", EXAMPLE_URL, "aNotDisplayed", "Cookies", "", -1);
+        ClientSideComponent four =
+                new ClientSideComponent(Map.of(), "A", "foo", "", EXAMPLE_URL, "aLink", "", "", -1);
+        // When
+        SortedSet<ClientSideComponent> sortedComponents =
+                new TreeSet<>(Set.of(one, two, three, four));
+        // Then
+        assertThat(sortedComponents, contains(three, one, four, two));
+    }
+
+    @Test
+    void shouldOrderByTypeForDisplayThenHrefThenTextDifferentHrefs() {
+        // Given
+        ClientSideComponent one =
+                new ClientSideComponent(
+                        Map.of(), "", "foo", "", EXAMPLE_URL, "zNotDisplayed", "Cookies", "", -1);
+        ClientSideComponent two =
+                new ClientSideComponent(Map.of(), "A", "foo", "", EXAMPLE_URL, "zLink", "", "", -1);
+        ClientSideComponent three =
+                new ClientSideComponent(
+                        Map.of(),
+                        "",
+                        "foo",
+                        "",
+                        "https://zoo.com",
```

Comment:
Use example.com as base domain.

Reply:
There are still more that need to be changed e.g. `foo.bar`.

Reply:
![image](https://github.com/user-attachments/assets/c6987dfa-af88-48fe-b906-b200219d8efe)

Reply:
Yes, this was addressed after and why is marked as resolved already.